### PR TITLE
Issue 2212

### PR DESCRIPTION
**Pull request by xiphmont**
Two patches implement the missing infrastructure needed to fix issue 2212 and allow proper Tile/LRU sizing in arbitrary 4:2:2 video.

### CODE REVIEW

File: src/tiling/tiler.rs [hunk 1]
```diff
@@ -85,9 +87,28 @@ impl TilingInfo {
         .ceil() as usize,
     );
 
-    let tile_cols_log2 =
+    let mut tile_cols_log2 =
```
Comment (barrbrain) [inline src/tiling/tiler.rs:90]: Nit, this mut is unnecessary.

File: src/header.rs [hunk 2]
```diff
@@ -671,25 +672,70 @@ impl<W: io::Write> UncompressedHeader for BitWriter<W, BigEndian> {
...
     let ti = &fi.tiling;
 
-    let cols_ones = ti.tile_cols_log2 - ti.min_tile_cols_log2;
-    for _ in 0..cols_ones {
-      self.write_bit(true);
-    }
-    if ti.tile_cols_log2 < ti.max_tile_cols_log2 {
-      self.write_bit(false);
-    }
+    if ((fi.sb_width + (1 << ti.tile_cols_log2) - 1) >> ti.tile_cols_log2)
```
Comment (barrbrain) [inline src/header.rs:683]: With `use crate::util::Fixed;` we can name this pattern:
```suggestion
    if fi.sb_width.align_power_of_two_and_shift(ti.tile_cols_log2)
```

File: src/header.rs [hunk 3]
```diff
@@ -671,25 +672,70 @@ impl<W: io::Write> UncompressedHeader for BitWriter<W, BigEndian> {
...
+      == ti.tile_width_sb
+      && ((fi.sb_height + (1 << ti.tile_rows_log2) - 1) >> ti.tile_rows_log2)
```
Comment (barrbrain) [inline src/header.rs:685]:
```suggestion
      && fi.sb_height.align_power_of_two_and_shift(ti.tile_rows_log2)
```

File: src/tiling/tiler.rs (hunk 1 above)
Comment (xiphmont) [reply src/tiling/tiler.rs]: done

File: src/header.rs (hunk 2 above)
Comment (xiphmont) [reply src/header.rs]: done

File: src/header.rs (hunk 3 above)
Comment (xiphmont) [reply src/header.rs]: done

**Pull request merged**